plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RewriteCast.java

```diff
@@ -66,6 +74,11 @@ private boolean pushdownSupported(JdbcTypeHandle sourceType, Type targetType)
         return switch (targetType) {
             case SmallintType _, IntegerType _, BigintType _ ->
                     SUPPORTED_SOURCE_TYPE_FOR_INTEGRAL_CAST.contains(sourceType.jdbcType());
+            // varchar -> char is unsupported as varchar supports multi-byte characters whereas char supports only single byte characters.
+            case CharType _ -> CHAR == sourceType.jdbcType();
+            // char -> varchar is not supported as Redshift doesn't pad char value with blanks whereas Trino pads char value with blanks.
```

Comment:
So even in case of performing cast between char(10) to char(100), does redshift pads whitespace to the resultant value ?

Reply:
you mean Trino or Redshift? Redshift doesn't pad blank character.

Reply:
I meant in Trino. Just to ensure that we don't read data more than without cast pushdown today

Reply:
This statement is contradictory to https://github.com/trinodb/trino/pull/23808/files#r1838491848 - So if we cast char(10) -> char(100) and if the char is `a` does it pad spaces for 99 times when we are trying to read the data from Redshift ? In that case the cast pushdown could leave us in reading more data rather than expected right ?